src/concrete_intersection.jl

```diff
@@ -339,33 +339,32 @@ This implementation unifies the constraints of the two sets obtained from the
 """
 function intersection(P1::AbstractPolyhedron{N},
                       P2::AbstractPolyhedron{N};
-                      backend=nothing,
-                      use_polyhedra_interface=false) where {N<:Real}
-    HPOLY = (P1 isa AbstractPolytope || P2 isa AbstractPolytope) ?
-        HPolytope{N} : HPolyhedron{N}
+                      backend=GLPKSolverLP()) where {N<:Real}
+
+    # if one of P1 or P2 is bounded => the result is bounded
+    HPOLY = (P1 isa AbstractPolytope || P2 isa AbstractPolytope) ? HPolytope{N} : HPolyhedron{N}
 
     # concatenate the linear constraints
     Q = HPOLY([constraints_list(P1); constraints_list(P2)])
 
     # remove redundant constraints
-    if use_polyhedra_interface
-        if backend == nothing
-            backend = default_polyhedra_backend(P1, N)
-        end
-        # convert to an hrep, remove the redundancies and convert back to HPOLY
-        ph = polyhedron(Q; backend=backend)
-        removehredundancy!(ph)
-        return convert(HPOLY, ph)
-    else
-        if backend == nothing
-            backend = GLPKSolverLP()
-        end
-        # here, detection of empty intersection may be reported as an infeasible LP
+    if backend isa AbstractMathProgSolver
```

Comment:
I am not sure that this is the correct criterion to distinguish between `Polyhedra`, but let's go for it.
In the `else` case we could check for [`isa Polyhedra.Library`](https://github.com/JuliaPolyhedra/Polyhedra.jl/blob/fc51bb5efdb4270ef8ff71561bf07bedc63ece0d/src/Polyhedra.jl#L10) and throw an error otherwise.

Reply:
> In the else case we could check for isa Polyhedra.Library and throw an error otherwise.

But i would need `using Polyhedra: Library` which requires Polyhedra. Is there some workaround around this?

Reply:
Oh, I see. You could outsource to another method that you then override with `Requires`, but that is much too complicated for a nicer error message. Let's keep it as is!

Reply:
In the last commit i added a comment about this issue.